### ISSUE TITLE
Segfault when hovering over settings button

### ISSUE THREAD

**Issue by FriesischScott** [haiwen/seafile-client#1379]
I am trying to package seafile-client 8.0.4 for Solus.

It builds fine, but segfaults when I hover over the new settings button in the main panel.

Valgrind produces the following error:
```
==15597== Process terminating with default action of signal 11 (SIGSEGV)
==15597==  Access not within mapped region at address 0x20
==15597==    at 0x5098E10: ??? (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x50657AF: QCommonStyle::drawControl(QStyle::ControlElement, QStyleOption const*, QPainter*, QWidget const*) const (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x50964F9: ??? (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x517DEFD: QPushButton::paintEvent(QPaintEvent*) (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x50249DD: QWidget::event(QEvent*) (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x4FE30AE: QApplicationPrivate::notify_helper(QObject*, QEvent*) (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x5ECCA69: QCoreApplication::notifyInternal2(QObject*, QEvent*) (in /usr/lib64/libQt5Core.so.5.15.3)
==15597==    by 0x501CB05: QWidgetPrivate::sendPaintEvent(QRegion const&) (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x501D339: QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x501E6D6: QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x501D043: QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib64/libQt5Widgets.so.5.15.3)
==15597==    by 0x501E6D6: QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib64/libQt5Widgets.so.5.15.3)
```

Any help would be appreciated.

**Comment by konstantingoretzki** [comment]
Can confirm this behavior on Arch Linux with the [seafile-client](https://aur.archlinux.org/packages/seafile-client) AUR package.

My valgrind output:
```
==19730== Process terminating with default action of signal 11 (SIGSEGV): dumping core
==19730==  Access not within mapped region at address 0x20
==19730==    at 0x5063810: ??? (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x502F24F: QCommonStyle::drawControl(QStyle::ControlElement, QStyleOption const*, QPainter*, QWidget const*) const (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x5060E4A: ??? (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x514B522: QPushButton::paintEvent(QPaintEvent*) (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x4FEE0BD: QWidget::event(QEvent*) (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x4FAAD61: QApplicationPrivate::notify_helper(QObject*, QEvent*) (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x5E993B9: QCoreApplication::notifyInternal2(QObject*, QEvent*) (in /usr/lib/libQt5Core.so.5.15.2)
==19730==    by 0x4FE5CF7: QWidgetPrivate::sendPaintEvent(QRegion const&) (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x4FE655B: QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x4FE7A0F: QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x4FE626C: QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib/libQt5Widgets.so.5.15.2)
==19730==    by 0x4FE7A0F: QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) (in /usr/lib/libQt5Widgets.so.5.15.2)
```

**Comment by killing** [comment]
Thank you for pointing out. We'll fix it in the next version.

**Comment by FriesischScott** [comment]
Should be fixed by #1384
